ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/UserManagementServiceIT.scala

```diff
@@ -109,71 +122,199 @@ final class UserManagementServiceIT extends LedgerTestSuite {
+          .mustFail("granting right to a non-existent user")
+        res3 <- ledger.userManagement.grantUserRights(
+          GrantUserRightsRequest("user1", List(adminPermission))
+        )
+        res4 <- ledger.userManagement.grantUserRights(
+          GrantUserRightsRequest("user1", userRightsBatch)
+        )
+      } yield {
+        assertSameElements(res1.newlyGrantedRights, List(adminPermission))
+        assertUserNotFound(res2)
+        assertSameElements(res3.newlyGrantedRights, List.empty)
+        assertSameElements(res4.newlyGrantedRights, userRightsBatch)
+      }
+    }
+
+    def testRevokeUserRights(): Future[Unit] = {
+      for {
+        res1 <- ledger.userManagement.revokeUserRights(
+          RevokeUserRightsRequest("user1", List(adminPermission))
+        )
+        res2 <- ledger.userManagement
+          .revokeUserRights(RevokeUserRightsRequest("user2", List(adminPermission)))
+          .mustFail("revoking right from a non-existent user")
+        res3 <- ledger.userManagement.revokeUserRights(
+          RevokeUserRightsRequest("user1", List(adminPermission))
+        )
+        res4 <- ledger.userManagement.revokeUserRights(
+          RevokeUserRightsRequest("user1", userRightsBatch)
+        )
+      } yield {
+        assertEquals(res1, RevokeUserRightsResponse(List(adminPermission)))
+        assertUserNotFound(res2)
+        assertSameElements(res3.newlyRevokedRights, List.empty)
+        assertSameElements(res4.newlyRevokedRights, userRightsBatch)
+      }
+    }
+
+    def testListUserRights(): Future[Unit] = {
+      for {
+        res1 <- ledger.userManagement.createUser(
+          CreateUserRequest(Some(User("user4", "party4")), Nil)
+        )
+        res2 <- ledger.userManagement.listUserRights(ListUserRightsRequest("user4"))
+        res3 <- ledger.userManagement.grantUserRights(
+          GrantUserRightsRequest(
+            "user4",
+            List(adminPermission, actAsPermission1, readAsPermission1),
+          )
+        )
+        res4 <- ledger.userManagement.listUserRights(ListUserRightsRequest("user4"))
+        res5 <- ledger.userManagement.revokeUserRights(
+          RevokeUserRightsRequest("user4", List(adminPermission))
+        )
+        res6 <- ledger.userManagement
+          .listUserRights(ListUserRightsRequest("user4"))
+      } yield {
+        assertEquals(res1, User("user4", "party4"))
+        assertEquals(res2, ListUserRightsResponse(Seq.empty))
+        assertSameElements(
+          res3.newlyGrantedRights,
+          Set(adminPermission, actAsPermission1, readAsPermission1),
+        )
+        assertSameElements(
+          res4.rights,
+          Set(adminPermission, actAsPermission1, readAsPermission1),
+        )
+        assertSameElements(res5.newlyRevokedRights, Seq(adminPermission))
+        assertSameElements(res6.rights, Set(actAsPermission1, readAsPermission1))
+      }
+    }
+
+    for {
+      _ <- testCreateUser()
```

Comment:
I think it would be better to have a separate test for each of the rpcs
It should be easy to achieve once you start using test case isolation mechanism as suggested above

Reply:
Ok